### ISSUE TITLE
[Feature Request] Overscan

### ISSUE THREAD

**Issue by Toadfield** [citra-emu/citra#5794]
There are small black borders on the left and the right,it would be so cool if you could just overscan it.

**Comment by poudink** [comment]
I'm not sure overscan means what you think it means.

**Comment by Toadfield** [comment]
I mean that you could overscan the black borders.

**Comment by TollyThaWally** [comment]
What do you mean by "overscan it"? Like poudink said, I don't think overscan means what you think it means

**Comment by Toadfield** [comment]
For what example the mesen emulator means with "overscan",that you can zoom in the image,so if it has black borders that you can make them away without stretching it.

**Comment by TollyThaWally** [comment]
Mesen is a NES emulator, the NES was designed to be used with a CRT TV which would automatically [overscan](https://en.wikipedia.org/wiki/Overscan) the video input from the NES, which Mesen can emulate by removing the edges of the screen like a TV would with a real console. The 3DS was never designed to be plugged into a TV, it had its own displays, so overscan doesn't apply to it. You can't just remove black borders by magic, they exist because the 3DS aspect ratio is not the same as your monitors, it is impossible to remove them without stretching the image.

**Comment by Toadfield** [comment]
That is not what I mean,for example there is this loading zone thing for all the tiles in smb3 on the right,with overscan you can select that the image gets zoomed in a little bit,so you can't see the things on the right anymore (but you will also see less on the left and up and down).
This should work with anything.

**Comment by Toadfield** [comment]
Some emulators call it like that for some reason

**Comment by TollyThaWally** [comment]
I don't get what your point is, the 3DS is not a NES, it was not designed for overscan so it would serve no purpose on a 3DS emulator

**Comment by Toadfield** [comment]
We are talking about something completely different,mesen or GlideN64 have an option that is called "overscan",it zooms into the whole picture.
This has nothing to do with the overscan that you mean.
The overscan that I mean should work with citra,it zooms into the picture how much you like until it is so much zoomed in that you can't see the black borders (but also not everything on the top and bottom,cause it is zoomed in),I hope that explanation made sense,I sadly can't find a video about the overscan that I mean.

**Comment by Dragios** [comment]
That is call **Zoomed to filled** feature like in YouTube Mobile. That is not *overscan*. You just caused confusion for other people by using the wrong term. Overscan is meant to hide artifacts exist in borders of the old analog TVs by shaving of those edges of the border, and as a result in does result in a little zoom in picture by at most 10% as a safe margin.

Overscan feature was never meant to remove black borders, fyi

**Comment by Toadfield** [comment]
But why do other emulators call it "overscan"?
Mesen and GlideN64 call it like that,sorry I thought that was the right defintion.
But that doesn't matter,would Zoomed to filled be easy to implement?

**Comment by Dragios** [comment]
Like others have mentioned, the other emulators you used or encounter uses overscan because the input video was meant for old analogue TV that uses CRT. Overscan feature is not needed for new display tech like LCD, although some LCD still have that feature for old compatibilities sake. Any console that is designed with modern display technologies (like LCD, LED and etc) will not have overscan feature at all

**Comment by Toadfield** [comment]
That is not what I mean,those emulators have a zoom in option that they call "overscan".

**Comment by Dragios** [comment]
@Toadfield The keyword here is video input signal that catered for CRT TV. The emulators you mentioned are all retro emulators that hook up to TV. Try using newer emulators like PS3 emu (rpcs3) or Switch emu such as _yuzu_ or _Ryujinx_, you will never find that feature call overscan.

Even handheld emulators like DesMuME (DS emu), Visual Boy Advance (GBA emu) you will not find overscan.

Please educate yourself 1st what does overscan do and its purpose of existance
https://www.avforums.com/threads/please-explain-overscan.855380/post-8055894

**Comment by Toadfield** [comment]
Thanks for explaining that

**Comment by Toadfield** [comment]
But what is with Zoomed to filled,would that be easy to implement?

**Comment by Dragios** [comment]
That is up to Citra developers. But from programmings perspective it should not be too hard to implement actually, basically just crop off extra screen resolution in either side so black borders does not appear as oppose to stretching the video, although both accomplish the same purpose but will yield different result.

Too bad the core Citra developers are all busy with their own stuff. Basically the development is on hiatus since Sept 2020. Not much happening ever since, only minor bug fixes. Some core developer here already migrated and focus on yuzu (Switch emulator) instead.

I could be wrong, they might have work behind the scenes which I have no idea of, but that's basically it, don't put your hopes too high is all I can say.

**Comment by vitor-k** [comment]
As mentioned above, the 3ds has LCD displays and so, like most handhelds or modern systems, does not have to deal with CRT quirks like overscan.
On consoles like the NES or the N64, the image will have added black bars or an extended display area to compensate for overscan, which is why it makes sense for emulators of said consoles to simulate overscan.
This fact also makes it safe for these emulators to crop the image that way, because the games will keep any essential parts of the picture away from the borders.
Given the 3ds does not have such considerations, cropping the image in any way will inevitably hide critical content like UI elements, text, and character sprites.
Simply having this feature would do more harm than good, as the number of users who will inadvertently enable it will be far greater than those who might use it.

You may be associating the overscan feature with "remove black borders", but the black borders on Citra have a different origin than the ones on said consoles/emulators.
The two screens the 3ds has have the resolutions of 400x240 for the top screen and 320x240 for the bottom screen. Therefore, when stacking them vertically, you have a 80 pixel difference between the two, and that's where the black bar comes from. Other black bars might appear when using other layouts and when in fullscreen with the "single screen" layout, as the screens are 15:9 and 12:9, so wouldn't fit perfectly on a 16:9 screen.

(obs: technically the top screen is 800x240 with non-square pixels for displaying the 3d effect, but Citra will only show the image for one of the eyes in normal circumstances)

@Dragios while it's true that development has slowed a lot the past year, I wouldn't say it's in hiatus. Not sure what you're basing the september date on, but since then there was the tomodachi life voice fix, merged in november, and the Pilotwings texture cache invalidation PR, merged in February. Not sure why you're talking about the lack of developers, this isn't really the place, and I don't see this being implemented even if there was an excess of development manpower.

**Comment by Dragios** [comment]
@vitor-k The September 2020 date was based on the last progress report in Citra website. My meaning of basically is _sort of_.
The fact that you can count only 2 meaningful PRs for the span of 9 months is proof enough that improvement on Citra emulator is sort of non-existant.

I have been following Citra's development when it is still in very early state since early 2016. I know there will be time where development would be more active during certain period of month. I have seen a lot of times where development slow a lot each year, but not until this degree if I were to compare to its previous years. I would say this year is the least productive year for Citra.

I wouldn't say there is lack of developers, just lack of developers pouring time to code here anymore. Each of them is busy with their own life now and some switch their time to focus on yuzu instead, which is a sad thing for me to say the least.

Yeah, and I apologized for saying this, but I agree it is not the right place to discuss this matter

**Comment by Toadfield** [comment]
@vitor-k
> Given the 3ds does not have such considerations, cropping the image in any way will inevitably hide critical content like UI elements, text, and character sprites.

It isn't that bad for every game,cropping the borders of doesn't hide ui elements in every game,so for some games it would work very good.
I mean the best thing would be a true widescreen option,but that would be much harder to implement.
